File: codeforces/119A/cpp/119a.cpp
#include <iostream>
using namespace std;

int gcd(int a, int b)
{
        if (b == 0)
                return a;

        return gcd(b, a % b);
}

int main()
{
        int n, a, b;
        cin >> a >> b >> n;

        int result = 1;
        int rm = 0;
        while (n > 0) {
                rm = gcd(result ? a : b, n);

                if (n < rm)
                        break;
                n -= rm;
                result = !result;
        }
        cout << result << endl;
        return 0;
}

File: codeforces/119A/py/119a.py
import sys

def gcd(a, b):
        if b == 0:
                return a
        return gcd(b, a % b)

a, b, n = map(int, sys.stdin.readline().split())
result = True
while n > 0:
        if result == True:
                rm = gcd(a, n)
        else:
                rm = gcd(b, n)

        if n < rm:
                break

        result = not result
        n -= rm

if result == True:
        print("1")
else:
        print("0")
